### DISCUSSION TITLE
Documentation and confusion

### DISCUSSION THREAD

**Discussion by binaryf** [coinfoundry/miningcore#1430] [General]
**Describe the bug**

Everything runs smoothly except "eth_sendTransaction".

According to https://github.com/oliverw/miningcore/wiki/Coin-Daemon-&-Wallet-RPC#ethereum
you can run go-ethereum (geth) with:

$ geth --cache 1024 --mine --http --http.port 8545 --unlock <your-wallet-address> --password <password-file-path> --allow-insecure-unlock

Mining works fine. 
But after a mined block, I get an error when Miningcore should reward the miners on the pool.

Miningcore says "System.Exception: eth_sendTransaction returned error: invalid sender code -32000"
Geth says err="invalid sender"

I can transfere founds in the go-ethereum javasript console.

Have I forgot something here ?

**To Reproduce**
1. Setup go-ethereum according to documentation.
2. Setup Miningcore according to documentation.
3. Run mining on Miningcore.
4. Find a block.
5. No payment for miners due to error in configuration.

**Expected behavior**

Mining and payment should work by following instructions or manual.

**Screenshots**

I can provide screenshot If needed, or any other information.
Maybe this is something that others also run into,
and can be fixed by providing a little more information.

**Platform and software**
 - Ubuntu 22.04 LTS
 - Latest Miningcore.
 - Go-ethereum (geth) v1.20.1-beta (https://github.com/ethereumpow/go-ethereum/releases)

**Comment by oliverw** [comment]
Forgive my ignorance but isn't this question purely academic as of yesterday?

**Comment by binaryf** [comment]
Maybe it is a bug ?

**Comment by oliverw** [comment]
> Maybe it is a bug ?

Its most likely a bug but what is the relevance after yesterdays merge?

**Comment by binaryf** [comment]
Ah! Got it! :)
Have a pool 90% set up for ETHPOW fork :)

**Comment by oliverw** [comment]
Ah ok. Will look into it in a bit.

**Comment by binaryf** [comment]
Let me know if you need more info.
Thanks!

**Comment by binaryf** [comment]
We also had an issue with miningcore when miningcore tried to pay miners more than the incoming balance in the base wallet.

It is possible that this was an issue (behaviour) beacause of a little messy dev-setup and the back and forth trying to get things working.

But it is very strange that miningcore are reporting reward: 5 when the ethereum pow block explorer says about 2 ether.

Here is the link to the dev setup: http://miningpool.no:4000/api/pools/miningpool1/blocks

UI is at http://miningpool.no/

You might want to look into this if this is related to the ethereum pow fork witch launches soon.

**Comment by oliverw** [comment]
So it was an uncle identified as a full block?

**Comment by binaryf** [comment]
I need to see the block explorer to be sure.

https://iceberg.ethwscan.com/blocks

But it is down at the moment.

**Comment by binaryf** [comment]
Looks like the block explorer for the testnet is permanently down.

It should be easy to reproduce. I can make a compact list of steps.

**Comment by binaryf** [comment]
Block reward should be 2.00 ETHW.

**Comment by totom9000** [comment]
Work Great here.   Its actualy  2ethw +netfee so it can be higer and you have unicles that is less  and orphan  0.

**Comment by oliverw** [comment]
> Block reward should be 2.00 ETHW.

What coin are we actually talking about?

**Comment by binaryf** [comment]
Ethereum POW (ETHPOW, ETHW)
https://ethereumpow.org/

**Comment by oliverw** [comment]
> Ethereum POW (ETHPOW, ETHW) https://ethereumpow.org/

So this is an issue about a coin that's not actually supported by MC? And you didn't even mention that?

**Comment by binaryf** [comment]
Sorry. I should have. 
Thought since it is basicly ethereum i did not underline it.

I thought the issue was I/O related between MC and Geth.

**Comment by seperatis** [comment]
> Ah! Got it! :) Have a pool 90% set up for ETHPOW fork :)

What was the solution for your issue?

`System.Exception: eth_sendTransaction returned error: invalid sender code -32000"
Geth says err="invalid sender"
`
I remember seeing the same error a very long time ago while testing ETH, but cannot remember what the solution was.

What was it in your case?

**Comment by oliverw** [comment]
Might be worth giving `dev` a shot. I have pushed 7973a2bf19a5acdddd57b1e88df3eebd4dd8d663 a couple hours ago.

**Comment by binaryf** [comment]
Great! Thank you.

**Comment by oliverw** [comment]
Any findings?

**Comment by binaryf** [comment]
New server is up. But the blockchain is in "heal mode".
Will let you know.

**Comment by oliverw** [comment]
> New server is up. But the blockchain is in "heal mode". Will let you know.

So?

**Comment by binaryf** [comment]
Still "State heal in progress"

**Comment by oliverw** [comment]
> Still "State heal in progress"

It really shouldn't take that long. I'd start over with a clean slate if I was you.

**Comment by totom9000** [comment]
Ethw actualy  takes manny manny hours to sync. 
We usualy dl chain add it inn manual as its multiple GB in size. And do a reindex as its faster. 
Ethw is same as eth. Etc  dont alow you to runn wallet in prune. And dont realy help woth good nodes as it simple go slow.

**Comment by oliverw** [comment]
> Ethw actualy takes manny manny hours to sync. We usualy dl chain add it inn manual as its multiple GB in size. And do a reindex as its faster. Ethw is same as eth. Etc dont alow you to runn wallet in prune. And dont realy help woth good nodes as it simple go slow.

Hours? @binaryf posted his last update 7 days ago.

**Reply by totom9000** [reply]
Yea hours need  be prepared  even  if your servers has 10gbps connection  to use  4-12h on a fresh sync.  Problem isnt your server its peers dont simple push 1gbps exanple. Its in bytes  and if Lucky  kbps

**Comment by binaryf** [comment]
Yep. And state heal never completed due to too slow disk read and write operations.

**Comment by binaryf** [comment]
I had better luck on my workstation at home, but it will be hard to find a block now at current difficulty
